packages/eslint-config-godaddy-minimal/index.js

```diff
@@ -0,0 +1,24 @@
+module.exports = {
+  extends: [
+    require('./extends')('eslint-config-godaddy-react-flow')
+  ],
+  rules: {
+    'complexity': 0,
+    'consistent-return': 0,
+    'id-length': 0,
+    'indent': [2, 2, {
+      VariableDeclarator: { var: 2, let: 2, const: 3 },
+      flatTernaryExpressions: true,
+      MemberExpression: 'off'
+    }],
+    'max-params': 0,
```

Comment:
On the fence for this one... I see the argument, but forcing into an "options" object if the function has a large number of arguments is a good thing now that destructuring basically gives us named arguments

Reply:
Is there any real difference between these two, though?
```es6
function doStuff(foo, bar, baz, qux, cat, dog) {
  // ...
}

doStuff(foo, bar, baz, qux, cat, dog);

function doStuff2(foo, bar, baz, { qux, cat, dog }) {
  // ...
}

doStuff2(foo, bar, baz, { qux, cat, dog });
```
Destructuring means that argument count and the apparent number of arguments to a function are no longer strictly coupled; the count of arguments passed is not a good measure of the size or complexity of a function's interface.

Reply:
There's a big difference when it comes to value passing.

For example, I see stuff like this all the time: `someFunction(app, false, 0, 'test', shopperId);`  What does `false` stand for in this case? Or `0`? I have to look at the source code of that function to understand it, whereas `someFunction({ app, env: 'test', shopperId, recursive: false, maxRetries: 0 })` is much easier to read.

Reply:
Boolean arguments to a function are a code smell anyway, if you follow the "clean code" stuff. Whether or not this is in an object doesn't change the underlying problem: a function taking a boolean argument is explicitly doing two things, and functions whose interfaces are exposed should only do one thing. Putting this inside an options object hides the fact that this function's responsibility is creeping, and gives an easy way for other people to add even more to it.

Reply:
I think it's nonsense to say that booleans always imply that something should be multiple functions. What about two booleans? Is four similar-but-differently-named functions better than one with parameters? Three booleans or 8 named functions? Sometimes is the case, but sometimes booleans are the better option.

Reply:
And now we've gotten to the point where we're discussing what the best interface for something should be at the level of engineers, which is where all discussions about interfaces should take place. None of the questions you raise here can be addressed via syntactic analysis.

Reply:
Considering my example of unreadable function call versus more readable, I think having the linting rule remains a better idea than disabling it.

Reply:
Though it's probably rare, there's a chance wrapping params in an object where the object could change during execution could cause bugs:
```
const p = { one: 1 };

function sideEffect(params) {
    console.log(params.one);
    p.one = 2;
    console.log(params.one);
}

sideEffect(p); // renders 1 and then 2
```
Using destructuring, immutability and a few other things would also prevent that, but it's still another potential consequence of forcing param wrapping.

Reply:
`function doStuff2(foo, bar, baz, { qux, cat, dog }) {`
the horror!

Honestly, function signatures are the things easily catchable in code reviews. A huge function signature, or a monstrosity like that above would get -1'd pretty quick.

As this style guide is defined as a "minimal" guide, I'm +1 on removing that rule.